js/components/address.js

```diff
@@ -0,0 +1,298 @@
+      this.suggestionSelected = true;
+
+      const data = await this.addressField.getAddressComponents(option.id);
+
+      this.addressComponents = data;
+
+      this.addressField.set(option.label);
+      this.updateValue();
+
+      this.onChange();
+    },
+    extractAddressComponents: function(place) {
+      var addressData = {
+        streetNumber: '',
+        streetName: '',
+        city: '',
+        state: '',
+        country: '',
+        postalCode: ''
+      };
+
+      if (place.length) {
+        place.forEach(component => {
+          var types = component.types;
+
+          if (types.includes('street_number')) {
+            addressData.streetNumber = component.long_name;
+          }
+
+          if (types.includes('route')) {
+            addressData.streetName = component.long_name;
+          }
+
+          if (types.includes('locality')) {
+            addressData.city = component.long_name;
+          }
+
+          if (types.includes('administrative_area_level_1')) {
+            addressData.state = component.long_name;
+          }
+
+          if (types.includes('country')) {
+            addressData.country = component.long_name;
+          }
+
+          if (types.includes('postal_code')) {
+            addressData.postalCode = component.long_name;
+          }
+        });
+      }
+
+      return addressData;
+    },
+    assignValuesToSeparateFields: function(place, separateFieldsName) {
+      var addressData = this.extractAddressComponents(place);
+
+      separateFieldsName.forEach(field => {
+        field.value = addressData[field.key];
+      });
+    },
+    updateSelectedFieldsProperty: function(attr, value) {
+      if (attr === 'readonly') {
+        value = !value;
+      }
+
+      var fields = this.$parent.fields;
+      var selectedValues = Object.values(this.selectedFieldOptions);
+
+      fields.forEach(field => {
+        if (selectedValues.includes(field.name)) {
+          field[attr] = value;
+        }
+      });
+    },
+    updateDisabledOptions: function() {
+      var selectedValues = Object.values(this.selectedFieldOptions);
+
+      this.fieldOptions.forEach(option => {
+        if (selectedValues.includes(option.label)) {
+          option.disabled = true;
+        } else {
+          option.disabled = false;
+        }
+      });
+    },
+    resetOptionsOnSelectOne: function(key) {
+      if (this.selectedFieldOptions[key] === '') {
+        this.updateDisabledOptions();
+      }
+    },
+    initAutocomplete: async function(input, countryRestrictions) {
+      this.addressField = Fliplet.UI.AddressField(this.$refs.addressField);
+      await this.addressField.getAutocompleteSuggestions(input, countryRestrictions)
+        .then(suggestions => {
+          if (typeof this.value === 'object') {
+            this.addressSuggestions = [];
+            this.suggestionSelected = true;
+          } else {
+            this.addressSuggestions = suggestions;
+          }
+        });
```

Comment:
@armine-fliplet, you should never combine `await` with `then`.

 ```suggestion
      const suggestions = await this.addressField.getAutocompleteSuggestions(input, countryRestrictions);
            if (typeof this.value === 'object') {
            this.addressSuggestions = [];
            this.suggestionSelected = true;
          } else {
            this.addressSuggestions = suggestions;
          }
```

Reply:
@dwfliplet Done